README.md

```diff
@@ -63,1498 +104,468 @@ Delegating a capability is like the statement `message`. Task is akin to `messag
-
-## 6.1 Schema
-
-```ipldsch
-type Result union {
-  | any "ok"
-  | any "error"
-} representation keyed
-```
-
-## 6.2 Variants
-
-## 6.2.1 Success
-
-The success branch MUST contain the value returned from a successful [Task] wrapped in the `"ok"` tag. The exact shape of the returned data is left undefined to allow for flexibility in various Task types.
-
-```json
-{ "ok": 42 }
-```
-
-## 6.2.2 Failure
-
-The failure branch MAY contain detail about why execution failed wrapped in the "error" tag. It is left undefined in this specification to allow for [Task] types to standardize the data that makes sense in their contexts.
-
-If no information is available, this field SHOULD be set to `{}`.
-
-```json
-{
-  "error": {
-    "dev/reason": "unauthorized",
-    "http/status": 401
-  }
-}
-```
-
-## 7 Effect
-
-The result of an [Invocation] MAY include a request for further actions to be performed via "effects". This enables several things: a clean separation of pure return values from requesting impure tasks to be performed by the runtime, and gives the runtime the control to decide how (or if!) more work should be performed.
-
-Effects describe requests for future work to be performed. All [Invocation]s in an [Effect] block MUST be treated as concurrent, unless explicit data dependencies between them exist via promise [Pipelines]. The `fx` block contains two fields: `fork` and `join`.
-
-[Task]s listed in the `fork` field are first-class and only ordered by promises; they otherwise SHOULD be considered independent and equal. As such, atomic guarantees such as failure of one effect implying failure of other effects if left undefined.
-
-The `join` field describes an OPTIONAL "special" [Invocation] which instruct the [Executor] that the [Task] [Invocation] is a continuation of the previous Invocation. This roughly emulates a virtual thread which terminates in an Invocation that produces Effect without a `join` field.
-
-Tasks in the `fork` field MAY be related to the Task in the `join` field if there exists a Promise referencing either Task. If such a promise does not exist, then they SHOULD be treated as entirely separate and MAY be scheduled, deferred, fail, retry, and so on entirely separately.
-
-## 7.1 Schema
-
-```ipldsch
-# Represents a request to invoke enclosed set of tasks concurrently
-type Effects {
-  # Primary set of tasks to be invoked
-  fork      [&Invocation]
-  
-  # Additional task to be invoked with added semantics
-  # of representing a workflow execution continuation.
-  join      optional &Invocation
-}
+})
 ```
 
-## 7.2 Fields
-
-### 7.2.1 Forked Task Invocations
+### Agents
+[Agents]: #agents
 
-The OPTIONAL `fork` field, if present MUST be a list of an alphabetically ordered [Task] links. List MUST NOT not contain duplicate entries.
+#### Issuer
+[Issuer]: #issuer
 
-### 7.2.2 Joined Task Invocation
+The `iss` field MUST include the Issuer of the Invocation. This DID MUST match against the encoding signature.
```

Comment:
```suggestion
The `iss` field MUST include the Issuer of the Invocation. This DID URL MUST dereference to the public key which proofs the signature over the payload included in the envelope.
```
As I mentioned on the top-level spec, a `did:key` doesn't need a fragment identifier because it always references to exactly one public key, but this is rare, most DID methods require you to express a qualified DID URL (i.e., a DID plus `#key1` or `#assertionMethod`, etc.) to get a specific key out of the DID Document.

Reply:
Manually ported so that I wouldn't forget to fix the typo in the `suggestion`. Thanks!